### DISCUSSION TITLE
[Feature] Public password protected share and collaborate link

### DISCUSSION THREAD

**Discussion by toger5** [immich-app/immich#2655] [Feature Request]
### The feature

I would like to use immich to share an album with a group of ppl that do not have an account on my instance.
Additionally i would like to give them the option to upload images to this album.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You can already do this from the album sharing settings.

**Reply by toger5** [reply]
That is really cool. Thanks. It was not mentioned inside the docs.

**Reply by marianjanco** [reply]
How can I protect public album with password as stated in the title of this feature proposal "Public password protected..."? I couldn't find that option anywhere in sharing settings. When I share the album publicly with the share link, it is accessible instantly without any prompts, so it is accessible literally to anyone who gets the link.

**Reply by jrasm91** [reply]
The "password" is embedded in the url itself.

**Reply by marekhanzlik** [reply]
Hmm, i don't think it is as safe as written password - eg.: when i send link via facebook, facebook will automatically open the page and can scrape the photos .. if it was password protected - where you have to know and write the password - this would not be possible

**Reply by jrasm91** [reply]
Possibly. This was a conversation we had originally when we were planning the feature. There are pros and cons to prompting for a password vs embedding it in the link. The pros of an embedded link are easier to share, easier to access, and almost all of the same security benefits as the prompted password. For those reasons that was the option we went with and I do not believe there are current plans to add an additional password feature to the links.

**Reply by marekhanzlik** [reply]
As i get the ease of use point, i still think the privacy is valid reason to implement classic password protection.
Thats the reason we want to use something else than google photos or other big company solution, to have complete privacy. Many current apps try to show you at least preview when you send a link: slack, fb, teams, outlook etc..
And there are not that many options to share the link "securely", even email providers will scan the link and do who knows what with the content and i really don't want them to do that with our photos.

I get that you have many other features planned that have higher prio. But if somebody made a PullRequest would you let it be implemented?

**Reply by jrasm91** [reply]
Possibly? Thoughts @alextran1502?

**Reply by alextran1502** [reply]
Yes, if someone make a PR to add password to share link, it will be reviewed and accepted
